src/signals/settings/roles/containers/RoleFormContainer/components/RoleForm/__test__/RoleForm.test.js

```diff
@@ -82,15 +81,44 @@ describe('/signals/settings/roles/components/RoleForm', () => {
     });
   });
 
+  it('should handle submit flow when saving an new role', () => {
+    props.role = undefined;
+    const { getByTestId } = render(withAppContext(<RoleForm {...props} />))
+
+    const event = {
+      target: {
+        value: 'nieuwe behandelaars',
+      },
+    };
+    fireEvent.blur(getByTestId('rolesFormFieldName'), event);
+
+    fireEvent.click(getByTestId('submitBtn'), { preventDefault: jest.fn() });
+
+    expect(props.onSaveRole).toHaveBeenCalledWith({
+      name: 'nieuwe behandelaars',
+      permission_ids: [],
+    });
+  });
+
+  it('should not do submit when form is invalid', () => {
+    props.role.name = '';
```

Comment:
You're changing the `prop` var here. Better to use
```jsx
const role = { ...props.role, name: '' };
const propsWithNoRoleName = { ...props, role };
const { getByTestId } = render(withAppContext(<RoleForm {...propsWithNoRoleName} />));
```

Reply:
dat hoeft niet want in de beforeEach wordt hij elke keer de props gezet

Reply:
gedaan